user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/EditBatchDialog.java

```diff
@@ -0,0 +1,228 @@
+    batchNameField.setRequired(true);
+    // must contain at least one non-whitespace character and no leading/tailing whitespace.
+    batchNameField.setPattern("^\\S+(.*\\S)*$");
+    batchNameField.setErrorMessage(
+        "The batch name must not be empty. It must not start nor end with whitespace.");
+    batchNameField.setValue(batchName);
+    batchNameField.addValueChangeListener(this::onBatchNameChanged);
+
+    Button addRow = new Button();
+    addRow.setText("Add Row");
+    addRow.addClickListener(this::onAddRowClicked);
+    addRow.addClassName("add-batch-row");
+
+    Button removeLastRow = new Button();
+    removeLastRow.setText("Remove Row");
+    removeLastRow.addClickListener(this::onRemoveLastRowClicked);
+    removeLastRow.addClassName("remove-batch-row");
+
+    setHeaderTitle("Update Batch");
+    setResizable(true);
+
+    Div batchControls = new Div();
+    batchControls.addClassName("batch-controls");
+    batchControls.add(batchNameField);
+
+    Span userHelp1 = new Span("You are editing samples for batch");
+    Span experimentNameText = new Span(experimentName);
+    experimentNameText.addClassName("experiment-name");
+    Span userHelp2 = new Span("and experiment");
+    batchNameText = new Span(batchName);
+    batchNameText.addClassName("batch-name");
+    Div userHelpText = new Div(userHelp1, batchNameText, userHelp2, experimentNameText);
+    userHelpText.addClassName("user-help-text");
+
+    Div spreadsheetControls = new Div();
+    spreadsheetControls.addClassName("spreadsheet-controls");
+
+    Span rowControls = new Span();
+    rowControls.addClassName("row-controls");
+    rowControls.add(addRow, removeLastRow);
+
+    Span errorText = new Span("Unspecific Error message");
+    errorText.addClassName("error-text");
+    errorText.setVisible(false);
+
+    spreadsheetControls.add(rowControls, errorText);
+
+    add(batchControls,
+        userHelpText,
+        spreadsheetControls,
+        spreadsheet);
+
+    batchNameField.focus();
+
+    spreadsheet.addValidationChangeListener(
+        validationChangeEvent -> {
+          if (validationChangeEvent.isInvalid()) {
+            errorText.setText(validationChangeEvent.getSource().getErrorMessage());
+            errorText.setVisible(true);
+          } else {
+            errorText.setVisible(false);
+          }
+        });
+  }
+
+  private void onBatchNameChanged(
+      ComponentValueChangeEvent<TextField, String> batchNameChangedEvent) {
+    batchNameText.setText(batchNameChangedEvent.getValue());
+  }
+
+  private void onRemoveLastRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.removeLastRow();
+  }
+
+  private void onAddRowClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.addEmptyRow();
+  }
+
+  @Override
+  protected void onConfirmClicked(ClickEvent<Button> clickEvent) {
+    spreadsheet.validate();
+    if (spreadsheet.isInvalid()) {
+      return;
+    }
+    if (batchNameField.isInvalid()) {
+      return;
+    }
```

Comment:
NIT: Instead of hardcoding the validation you could also iterate over all childrens within this component and check for each children that extends the HasValidation interface if they're valid, then this does not have to be updated every time the component changes. 
Similiar how i i did it in the stepper(see below) 

```
 private boolean isDialogContentValid() {
    return dialogContent.getChildren().filter(component -> component instanceof HasValidation)
        .map(component -> ((HasValidation) component).isInvalid()).anyMatch(aBoolean -> !aBoolean);
  }
```

Reply:
that is done in `validate` ;)

Reply:
Yes but you have seperate calls for the Spreadsheet and the BatchnameField. What i meant was that the dialog checks its children and validates as necessary :)